FusionIIIT/applications/department/views.py

```diff
@@ -0,0 +1,149 @@
+import datetime
+import json
+from operator import or_
+from functools import reduce
+
+from django.contrib import messages
+from django.contrib.auth.decorators import login_required
+from django.http import HttpResponse, HttpResponseRedirect
+# Create your views here.
+from django.db.models import Q
+from django.shortcuts import render
+from django.contrib.auth.models import User
+from django.core.paginator import Paginator, EmptyPage, PageNotAnInteger
+from applications.academic_information.models import Spi, Student
+from applications.globals.models import (Designation, ExtraInfo,
+                                         HoldsDesignation)
+from applications.eis.models import (faculty_about, emp_research_projects)
+
+#from .models import ()
+from jsonschema import validate
+from jsonschema.exceptions import ValidationError
+
+# Create your views here.
+
+def index(request):
+    context = {}
+    print(request.user)
+    if(str(request.user)!="AnonymousUser"):
+        return HttpResponseRedirect('/dashboard/')
+    else:
+        return render(request, "globals/index1.html", context)
+
+@login_required(login_url='/accounts/login')
+def admin(request):
+    admin = Designation.objects.get(name='admin')
+    hod = Designation.objects.get(name='hod')
+    hd_admin = HoldsDesignation.objects.filter(
+        user=request.user, designation=admin)
+    hd_hod = HoldsDesignation.objects.filter(
+        user=request.user, designation=hod)
+
+
+def hod(request):
+    ###### not working need to find another way to implement
+    user = request.user
+    
+    #cse_hod = request.user.holds_designations.filter(designation__name='hod').exists()
+    fac_view = request.user.holds_designations.filter(designation__name='faculty').exists() 
+    print(request.user.holds_designations.filter(designation__name='faculty'))
+    student = request.user.holds_designations.filter(designation__name='student').exists()
+
+    # finding designation of user
+    user_designation = ""
+    # if cse_hod:
+    #     user_designation = "hod"
+    # elif fac_view:
+    #     user_designation = "faculty"
+    # elif student:
+    #     user_designation = "student"
+    
+    print(fac_view, student) 
+    if fac_view:
+        user_designation = "faculty"
+    elif student:
+        user_designation = "student"
+    if user_designation == "student":
+        return render(request,"department/index.html")
+    elif(str(user.extrainfo.user_type)=='faculty'):
+        return render(request, 'department/dep_request.html', {"user_designation":'faculty'})
+
+def file_request(request):
+    #return render(request, "department/dep_complaint.html")
+    return render(request, 'department/dep_request.html')
+
+def BtechFirstYear_Students(request):
```

Comment:
Please, it's been told several times to follow PEP-8. If you couldn't find that doc, here is the [link](https://www.python.org/dev/peps/pep-0008/)
Also, to summarize for you, take care of these points:-

- Make sure each view function and model class should have a docstring.
- Class name is in PascalCase.
- Functions, variables name should be in snake_case.